src/cryptonote_core/tx_pool.cpp

```diff
@@ -106,6 +106,36 @@ namespace cryptonote
   tx_memory_pool::tx_memory_pool(Blockchain& bchs): m_blockchain(bchs), m_txpool_max_size(DEFAULT_TXPOOL_MAX_SIZE), m_txpool_size(0)
   {
 
+  }
+  //---------------------------------------------------------------------------------
+  bool tx_memory_pool::have_deregister_tx_already(transaction const &tx) const
```

Comment:
Reject TX's passed around with a different combination of votes that already are waiting in the mempool.

Reply:
Do you also check previous blocks?